### PR TITLE
Validation scope fixes

### PR DESCRIPTION
**Pull request by chapcz**
I have no compiler for coffe script. Can you please meke it for me?

### CODE REVIEW

**Comment by chapcz** [comment]
Hi, something missing in this PR? I really want to fix this bug soon. :bug:

File: src/Column/MultiAction.php [hunk 1]
```diff
@@ -151,4 +157,30 @@ public function getTemplateVariables()
...
 		]);
 	}
+
+
+	/**
+	 * @param string $actionKey
+	 * @param callable $rowCondition
+	 * @return void
+	 */
+	public function setRowCondition($actionKey, callable $rowCondition)
```
Comment (paveljanda) [inline src/Column/MultiAction.php]: How are row conditions related to validation scope fix? :)

File: src/templates/column_multi_action.latte [hunk 2]
```diff
@@ -5,7 +5,9 @@
 <div class="dropdown">
 	{$multi_action->renderButton()|noescape}
 	<ul class="dropdown-menu">
-		<li n:foreach="$multi_action->getActions() as $action">
+		<li n:foreach="$multi_action->getActions() as $actionKey => $action">
```
Comment (paveljanda) [inline src/templates/column_multi_action.latte]: Again, how is that related to this topic (validation scope fix)?

**Comment by chapcz** [comment]
Hi, I continued in your branch for this topic. I can clean these lines in the evening.

**Comment by chapcz** [comment]
@paveljanda removed "multiaction" code

**Comment by paveljanda** [comment]
@chapcz It is not actually working. Inline edit/add with required input does not throw an alert.

**Comment by chapcz** [comment]
@paveljanda do you use non minified datagrid.js? For me it is working both (add and edit). In this PR is not included minified version (I dont know method which is used for minify)

**Comment by chapcz** [comment]
`

        $grid->addInlineEdit()
            ->onControlAdd[] = function(Container $container) {
            $container->addText('foo', '')->setRequired('Fill foo - edit');
            $container->addText('boo', '')->setRequired('Fill boo - edit');
         };

        $grid->addInlineAdd()
            ->onControlAdd[] = function(Container $container) {
            $container->addText('foo', '')->setRequired('Fill foo - add');
            $container->addText('boo', '')->setRequired('Fill boo - add')  ;
         };
`

image
image

**Comment by paveljanda** [comment]
@chapcz Aha, I used the datagrid.coffee file. Should I use only the datagrid.js?

**Comment by chapcz** [comment]
I used in mentioned example all files from bower (similary as documentation example) but only thing which I changed is datagrid.js file.

**Comment by paveljanda** [comment]
@chapcz What datagrid,js version are you using? When I checked out branch `validation-scope-fixes`, there was about 500 changed lines when comparing to datagrid.js file in your PR. Which is weird.

Anyway, what branch did you fork? The `validation-scope-fixes`? Can you make the PR back into this branch? Thanks a lot!

**Comment by chapcz** [comment]
@paveljanda ok, I changed the branch .... but I doun´t understand that you see 500 changed lines, but in PR  I see 15 changes

**Comment by chapcz** [comment]
Can I do something else in this PR?

**Comment by chapcz** [comment]
@paveljanda is there anything what can I do to move this issue forward?

**Comment by paveljanda** [comment]
I checked out this revision but it's not really working the way I suppose it should. The inline add form does not show any validation alerts.

**Comment by chapcz** [comment]
chm, I think we are back in 19 sep. In my case, I have almost emty nette project, inside I cloned https://github.com/chapcz/datagrid.git and checked out branch validation-scope-fixes. In template I use almost everything from current bower. The only thing which I use from this repository is datagrid.js (also min version). In my example if fires both - add and edit forms as I showed in examples . :-(

**Comment by paveljanda** [comment]
Ok, let me create also an empty project with dependency on this branch.

**Comment by chapcz** [comment]
Hi again. Today I updated my fork with base project and tested again. Validation works for me, only thing which is not working is  when I use min version of datagrid.js ... so can we make some progress in this please? Can you try this with non minified version please? I will try to do my best. Thanks.

**Comment by paveljanda** [comment]
Hi. Thanks for you work. I really appreciate it. I fixed it in a bit different way in upcoming version 6.

https://github.com/contributte/datagrid/commit/5e8db96fa74606b8f34b31e90ae867d113e60431
